pysqldb3/tests/test_io.py

```diff
@@ -241,159 +241,184 @@
-#         assert not db.table_exists(schema='working', table=test_sql_to_pg_qry_table)
-#
-#         # Add test_table
-#         sql.drop_table(schema='dbo', table=test_sql_to_pg_qry_table)
-#         sql.query("""
-#         create table dbo.{} (test_col1 int, test_col2 int);
-#         insert into dbo.{} VALUES(1, 2);
-#         insert into dbo.{} VALUES(3, 4);
-#         """.format(test_sql_to_pg_qry_table, test_sql_to_pg_qry_table, test_sql_to_pg_qry_table))
-#
-#         # sql_to_pg_qry
-#         data_io.sql_to_pg_qry(sql, db, query="select * from dbo.{}".format(test_sql_to_pg_qry_table),
-#                               dest_table=test_sql_to_pg_qry_table, dest_schema='working', print_cmd=True)
-#
-#         # Assert sql_to_pg_qry successful and correct length
-#         assert db.table_exists(schema='working', table=test_sql_to_pg_qry_table)
-#         assert len(db.dfquery('select * from working.{}'.format(test_sql_to_pg_qry_table))) == 2
-#
-#         # Assert df equality
-#         sql_df = sql.dfquery("""
-#         select * from dbo.{}
-#         order by test_col1
-#         """.format(test_sql_to_pg_qry_table)).infer_objects().replace('\s+', '', regex=True)
-#
-#         pg_df = db.dfquery("""
-#         select * from working.{}
-#         order by test_col1
-#         """.format(test_sql_to_pg_qry_table)).infer_objects().replace('\s+', '', regex=True)
-#
-#         sql_df.columns = [c.lower() for c in list(sql_df.columns)]
-#
-#         # Assert
-#         pd.testing.assert_frame_equal(sql_df, pg_df.drop(['ogc_fid', 'geom'], axis=1), check_column_type=False,
-#                                       check_dtype=False)
-#
-#         # Cleanup
-#         db.drop_table(schema='working', table=test_sql_to_pg_qry_table)
-#         sql.drop_table(schema='dbo', table=test_sql_to_pg_qry_table)
-#
-#     def test_sql_to_pg_qry_no_dest_table_input(self):
-#         return
-#
-#     def test_sql_to_pg_qry_empty_query_error(self):
-#         return
-#
-#     def test_sql_to_pg_qry_empty_wrong_layer_error(self):
-#         return
-#
-#     def test_sql_to_pg_qry_empty_overwrite_error(self):
-#         return
-#
-#     # Note: temporary functionality will be tested separately!
-#     # Still to test: LDAP, print_cmd
-#
-#
+class TestSqlToPgQry:
+    def test_sql_to_pg_qry_basic_table(self):
+        # Assert pg table doesn't exist
+        db.drop_table(schema='working', table=test_sql_to_pg_qry_table)
+        assert not db.table_exists(table=test_sql_to_pg_qry_table)
+
+        # Add test_table
+        # sql.drop_table(schema='dbo', table=test_sql_to_pg_qry_table)
+        # sql.query("""
+        # create table dbo.{} (test_col1 int, test_col2 int);
+        # insert into dbo.{} VALUES(1, 2);
+        # insert into dbo.{} VALUES(3, 4);
+        # """.format(test_sql_to_pg_qry_table, test_sql_to_pg_qry_table, test_sql_to_pg_qry_table))
+
+        # sql_to_pg_qry
+        data_io.sql_to_pg_qry(sql, db, query="""-- comments within the query
```

Comment:
if possible we shouldnt rely on tables being present for testing, you should be able to test with making a global temp table ie. `f'##{sql_table_name }`

Reply:
I kept trying to get this to work on Fri/weekend, and it won't insert data into a global temp table when I run the `sql_to_pg_qry` function. I can create the temp table and `select` it, but the moment I add `insert` to the global temp table, it won't create the table in PG (it doesn't error out when I run the function, but if I try to access the table in PG, it doesn't exist).

Tried looking online to figure out the error, but can't find any existing literature.  Doesn't work for local temp table either. Not sure if you might have any luck with getting it to work on your end, or we need to change how we test this function without calling an existing table.